pkg/reduce/pnf/processandforward.go

```diff
@@ -78,7 +85,6 @@ func NewProcessAndForward(ctx context.Context,
 func (p *ProcessAndForward) Process(ctx context.Context) error {
 	var err error
 
-	// FIXME: we need to fix https://github.com/numaproj/numaflow-go/blob/main/pkg/function/service.go#L101
 	// blocking call, only returns the result after it has read all the messages from pbq
 	p.result, err = p.UDF.ApplyReduce(ctx, &p.PartitionID, p.pbqReader.ReadCh())
```

Comment:
we also want to capture the duration for processing a partition including reading, waiting and invoking udf